### PR TITLE
chore: remove snapshot counters

### PR DESCRIPTION
**Pull request by arriqaaq**
## Description

The idea to keep track of active snapshots was to control the number of snapshots spawned in case we write snapshots to file. There is no need to keep track of snapshots currently as it provides no real benefit, and there are no writes to disk for vart right now.

### CODE REVIEW

**Review by RaphaelDarley — APPROVED** [review]
looks good

File: Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "vart"
 publish = true
-version = "0.1.1"
+version = "0.1.2"
```
Comment (phughk) [inline Cargo.toml:4]: Would still recommend version upgrades to be separate PRs :P But it's fine for now obvs - it can stay in

File: src/art.rs [hunk 2]
```diff
@@ -1171,66 +1148,17 @@ impl<P: KeyTrait, V: Clone> Tree<P, V> {
     /// Returns a `Result` containing the `Snapshot` if the snapshot is created successfully,
     /// or an `Err` with an appropriate error message if creation fails.
     ///
-    pub fn create_snapshot(&mut self) -> Result<Snapshot<P, V>, TrieError> {
+    pub fn create_snapshot(&self) -> Result<Snapshot<P, V>, TrieError> {
         // Check if the tree is already closed
-        self.is_closed()?;
```
Comment (phughk) [inline src/art.rs:1176]: Shouldnt this check stay?

**Review by phughk — APPROVED** [review]
Lgtm, thanks!

File: src/art.rs (hunk 2 above)
Comment (arriqaaq) [reply src/art.rs:1176]: You are right, I have actually just changed the name

File: Cargo.toml (hunk 1 above)
Comment (arriqaaq) [reply Cargo.toml:4]: Yes you are right in this

Comment (arriqaaq) [reply Cargo.toml:4]: Will do this from next time
